### DISCUSSION TITLE
Logically bound image Vs Standard quadlet

### DISCUSSION THREAD

**Discussion by sigulete** [bootc-dev/bootc#810] [General]
I have a server implemented using `fedora-bootc` that has been solid like a rock for the last 3 months.
Bootc is a great technology, and I'm happy to try new ideas, hence the reason to share what I understand about _logically bound images_ (technology preview).

My server runs several services implemented using Pods and Containers. As such this technology preview caught my attention.
I have tried to make a comparison between the current way of implementing containers and this new idea. But I'm not sure if the summary below is 100% accuate. And what other benefits _logically bound images_ would bring?

### Lifecycle

- **Standard Quadlet:** The images are downloaded by the (bootc based) server the first time it starts. And container updates are handled by the user as needed using `podman auto-update`
- **Logically bound image:** The images are included in the bootable container and available when the (bootc based) server starts. Container updates are handled by the container image and deployed as part of the upgrade (`bootc upgrade`).

### Upgrades

- **Standard Quadlet:** Managed by the user (`podman auto-update`). They can be triggered at anytime independent of the container image.
- **Logically bound image:** Managed by the container image, they are part of the base image. The user has no control

### Garbage collecton

- **Standard Quadlet:** It is up to the user to clean up old images (`podman image prune`)
- **Logically bound image:** Managed by the container image during upgrades. No action is required by the user 

### User-space container

- **Standard Quadlet:** It can be implemented
- **Logically bound image:** Not sure if this is possible !!

The final choice would depends on  each use-case.

**Comment by cgwalters** [comment]
I think you have it all right. Is there anything missing from the docs in this aspect you see? I think it's basically just "rootless" containers because we don't have a robust way to share those today, but in any case IMO one should generally [use --userns=auto](https://github.com/containers/podman/discussions/13728) today instead for most server use cases.

**Reply by sigulete** [reply]
The docs are OK. However, there looks to be a typo in the example with `ln -s /usr/share/containers/systemd/my-app.image`  

It is repeated twice, and I assume the second should reference to `another-app.image` (or `another-app.container`)

Also, if you consider appropriate, using the comparison with standard quadlet containers, please include it in the docs (or I can open a PR). It will make it easier to understand.

**Reply by cgwalters** [reply]
I put up https://github.com/containers/bootc/pull/811 which has some tweaks

I changed "standard" in your table to "floating" because terms like "default", "standard" and "normal" tend to change meaning over time.

I also condensed the table a bit and reworded things, let me know what you think!

**Reply by sigulete** [reply]
Looks good.
